src/engine/controls/cuecontrol.cpp

```diff
@@ -871,22 +1110,48 @@ void CueControl::hotcueClear(HotcueControl* pControl, double v) {
     m_pHotcueFocus->set(Cue::kNoHotCue);
 }
 
-void CueControl::hotcuePositionChanged(HotcueControl* pControl, double newPosition) {
+void CueControl::hotcuePositionChanged(
+        HotcueControl* pControl, double newPosition) {
     QMutexLocker lock(&m_mutex);
-    if (!m_pLoadedTrack)
+    if (!m_pLoadedTrack) {
         return;
+    }
 
     CuePointer pCue(pControl->getCue());
     if (pCue) {
         // Setting the position to Cue::kNoPosition is the same as calling hotcue_x_clear
         if (newPosition == Cue::kNoPosition) {
             detachCue(pControl);
         } else if (newPosition > 0 && newPosition < m_pTrackSamples->get()) {
+            //TODO: Handle Loops
             pCue->setStartPosition(newPosition);
         }
     }
 }
 
+void CueControl::hotcueEndPositionChanged(
+        HotcueControl* pControl, double newEndPosition) {
+    QMutexLocker lock(&m_mutex);
```

Comment:
Why lock m_mutex?

Reply:
Because it calls `detachCue`, which is also protected by the mutex. By the whole locking code is mess atm.